### PR TITLE
Add descriptions to MDX frontmatter

### PR DESCRIPTION
**Pull request by MaedahBatool**
The description was missing from all the docs pages. This PR has all these new changes. 

### Reference issues

- #576

### CODE REVIEW

**Comment by MaedahBatool** [comment]
Hi @leerob, can you please review this? Becca told me that you'd be the right one to do it. 🙈 Thank you...

File: docs/pages/terms.mdx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Terms of Service
+description: Read Turborepo's terms and conditions here with all the set rules and guidelines.
```
Comment (leerob) [inline docs/pages/terms.mdx:3]:
```suggestion
description: Learn about the Terms of Service for Turborepo.
```

File: docs/pages/privacy.mdx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Privacy Policy
+description: Read Turborepo's privacy policy here with all the set rules and guidelines.
```
Comment (leerob) [inline docs/pages/privacy.mdx:3]:
```suggestion
description: Read the Privacy Policy for Turborepo.
```

File: docs/pages/docs/upgrading-to-v1.mdx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Upgrading to v1
+description: Turborepo joins Vercel, after which our turbo CLI is now open-source, and remote caching is available on all accounts.
```
Comment (leerob) [inline docs/pages/docs/upgrading-to-v1.mdx:3]:
```suggestion
description: Learn how to upgrade to Turborepo v1, now open-source and part of Vercel.
```

File: docs/pages/docs/reference/configuration.mdx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Configuration Options
+description: Learn how to configure the Turborepo CLI by adding an object called turbo right from your package.json file.
```
Comment (leerob) [inline docs/pages/docs/reference/configuration.mdx:3]:
```suggestion
description: Learn how to configure Turborepo through `turbo.json`.
```

File: docs/pages/docs/reference/command-line-reference.mdx [hunk 5]
```diff
@@ -1,3 +1,8 @@
+---
+title: CLI Reference
+description: Make the most out of Turborepo's CLI. Install turbo, where you'll get a monorepo with apps (i.e., docs and web) and packages.
```
Comment (leerob) [inline docs/pages/docs/reference/command-line-reference.mdx:3]:
```suggestion
description: Turborepo is a high-performance build system for JavaScript and TypeScript codebases.
```

File: docs/pages/docs/reference/codemods.mdx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Turborepo Codemods
+description: To refactor large-scale codebases, Turborepo offers incredible codemod transformations and migration scripts every time some feature is removed.
```
Comment (leerob) [inline docs/pages/docs/reference/codemods.mdx:3]:
```suggestion
description: To make upgrading easier, Turborepo includes codemods and migration scripts.
```

File: docs/pages/docs/index.mdx [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Documentation
+description: Turborepo, developed by Jared Palmer and acquired by Vercel, offers a robust solution for JavaScript/TypeScript-based monorepos that stores all of your code and assets for every project.
```
Comment (leerob) [inline docs/pages/docs/index.mdx:3]:
```suggestion
description: Turborepo is a high-performance build system for JavaScript and TypeScript codebases.
```

File: docs/pages/docs/guides/migrate-from-lerna.mdx [hunk 8]
```diff
@@ -1,3 +1,8 @@
+---
+title: Migrate from Lerna
+description: Turborepo and Lerna have several features that are common to both. However, they are different in many ways as well. This quick and handy tutorial explains how to use Turborepo with Lerna.
```
Comment (leerob) [inline docs/pages/docs/guides/migrate-from-lerna.mdx:3]:
```suggestion
description: Learn how to incrementally adopt Turborepo from your existing Lerna monorepo.
```

Feedback: If we say "However, they are different in many ways as well" - it's better to just explain _what_ those differences are, or don't mention lines like that 👍

File: docs/pages/docs/guides/complimentary-tools.mdx [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Awesome Monorepo Tools
+description: These are the list of tools that work tremendously well and are compatible when you're using with Turborepo.
```
Comment (leerob) [inline docs/pages/docs/guides/complimentary-tools.mdx:3]:
```suggestion
description: Versioning, publishing, changelogs, linting, testing, automation, and more.
```

File: docs/pages/docs/glossary.mdx [hunk 10]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Glossary
+description: A brief description of some of the technical terms you might encounter while using Turborepo.
```
Comment (leerob) [inline docs/pages/docs/glossary.mdx:3]:
```suggestion
description: A brief description of technical terms you might encounter while using Turborepo.
```

File: docs/pages/docs/getting-started.mdx [hunk 11]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Getting Started
+description: Learn how to get started with Turborepo. Create a new starter monorepo or add Turborepo to an existing project.
```
Comment (leerob) [inline docs/pages/docs/getting-started.mdx:3]:
```suggestion
description: Create your first monorepo or add Turborepo to an existing project.
```

File: docs/pages/docs/features/scopes.mdx [hunk 12]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Scoped tasks
+description: To ensure even faster builds, Turborepo's scope option allows the running tasks to proceed to the packages that match the scope argument.
```
Comment (leerob) [inline docs/pages/docs/features/scopes.mdx:3]:
```suggestion
description: Even faster Turborepo builds with scoped tasks.
```

File: docs/pages/docs/features/remote-caching.mdx [hunk 13]
```diff
@@ -1,5 +1,7 @@
 ---
 title: Remote Caching
+description: Turborepo shares a remote cache with your teammates and CI/CD for an additional speed boost. This is possible because it uses addressable content storage.
```
Comment (leerob) [inline docs/pages/docs/features/remote-caching.mdx:3]:
```suggestion
description: Share cache artifacts across machines for even faster builds.
```

File: docs/pages/docs/features/pipelines.mdx [hunk 14]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Pipelines
+description: Pipelining defines relationships between your tasks and lets Turborepo optimize what and when to build.
```
Comment (leerob) [inline docs/pages/docs/features/pipelines.mdx:3]:
```suggestion
description: Pipelines with Turborepo specify task relationships explicitly and conventionally.
```

File: docs/pages/docs/features/caching.mdx [hunk 15]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Caching
+description: Caching with Turporepo can save you a lot of time by emitting files and logs of previously run commands.
```
Comment (leerob) [inline docs/pages/docs/features/caching.mdx:3]:
```suggestion
description: Caching allows Turborepo to skip work that's already been done, for the fastest builds.
```

Nit: avoid phrases like "a lot of time" - how much time?

File: docs/pages/docs/changelog.mdx [hunk 16]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Changelog
+description: We add these modifications to the changelog file whenever some changes are made to the Turborepo.
```
Comment (leerob) [inline docs/pages/docs/changelog.mdx:3]:
```suggestion
description: Hear about new feature additions and changes to Turborepo.
```

File: docs/pages/docs/acknowledgments.mdx [hunk 17]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Acknowledgements and Prior Art
+description: A big thank you to all these developers, build systems, and monorepo tools for the immense support and help which made the development of Turborepo possible.
```
Comment (leerob) [inline docs/pages/docs/acknowledgments.mdx:3]:
```suggestion
description: Thank you to all these developers, build systems, and monorepo tools for their support and assistance.
```

Nit: "A big" - Can be more direct -> Just go straight to "Thank you"

**Review by leerob — DISMISSED** [review]

**Comment by MaedahBatool** [comment]
@gsoltis please can we get this PR merged? 🙌

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**